`azurerm_[linux|windows]_virtual_machine`: Add support for `dedicated_host_group_id` (#14936)
hashicorp/terraform-provider-azurerm: website/docs/r/linux_virtual_machine.html.markdown

* `custom_data` - (Optional) The Base64-Encoded Custom Data which should be used for this Virtual Machine. Changing this forces a new resource to be created.

* `dedicated_host_id` - (Optional) The ID of a Dedicated Host where this machine should be run on. Conflicts with `dedicated_host_group_id`.

> * `dedicated_host_group_id` - (Optional) The ID of a Dedicated Host Group where this machine should be run on. The target Host Group must have automatic placement enabled. Conflicts with `dedicated_host_id`.

Reviewer: ```suggestion
* `dedicated_host_group_id` - (Optional) The ID of a Dedicated Host Group that this Linux Virtual Machine should be run within. Conflicts with `dedicated_host_id`.

-> **Note:** The Dedicated Host Group must have automatic placement enabled.
```

Author: Updated
